src/main/java/com/uber/cadence/internal/sync/SyncWorkflowWorker.java

```diff
@@ -115,64 +136,80 @@ public void setLocalActivitiesImplementation(Object... activitiesImplementation)
     this.laTaskHandler.setLocalActivitiesImplementation(activitiesImplementation);
   }
 
+  public void setActivitiesImplementationToDispatchLocally(Object... activitiesImplementation) {
+    this.ldaTaskHandler.setActivitiesImplementation(activitiesImplementation);
+  }
+
   @Override
   public void start() {
     workflowWorker.start();
     // workflowWorker doesn't start if no types are registered with it. In that case we don't need
     // to start LocalActivity Worker.
     if (workflowWorker.isStarted()) {
       laWorker.start();
+      ldaWorker.start();
     }
   }
 
   @Override
   public boolean isStarted() {
-    return workflowWorker.isStarted() && laWorker.isStarted();
+    return workflowWorker.isStarted() && laWorker.isStarted() && ldaWorker.isStarted();
   }
 
   @Override
   public boolean isShutdown() {
-    return workflowWorker.isShutdown() && laWorker.isShutdown();
+    return workflowWorker.isShutdown() && laWorker.isShutdown() && ldaWorker.isShutdown();
   }
 
   @Override
   public boolean isTerminated() {
-    return workflowWorker.isTerminated() && laWorker.isTerminated();
+    return workflowWorker.isTerminated()
+        && laWorker.isTerminated()
+        && ldaHeartbeatExecutor.isTerminated()
+        && ldaWorker.isTerminated();
   }
 
   @Override
   public void shutdown() {
     laWorker.shutdown();
+    ldaHeartbeatExecutor.shutdown();
```

Comment:
Why are we shutting down `ldaHeartbeatExecutor` but not the `heartbeatExecutor`? How do they differ?

Reply:
yeah, I had decided to test it in the separate diff to be safe.